src/main/java/com/github/manolo8/darkbot/gui/MainGui.java

```diff
@@ -46,6 +70,26 @@ public MainGui(Main main) throws HeadlessException {
         toFront();
         requestFocus();
         setAlwaysOnTop(main.config.BOT_SETTINGS.DISPLAY.ALWAYS_ON_TOP);
+
+        addComponentListener(new ComponentAdapter() {
+            @Override
+            public void componentMoved(ComponentEvent e) {
+                if (botSettings.SAVE_MAIN_GUI_POS) {
+                    botSettings.mainGuiX = getX();
+                    botSettings.mainGuiY = getY();
+                    ConfigEntity.changed();
+                }
+            }
+
+            @Override
+            public void componentResized(ComponentEvent e) {
+                if (botSettings.SAVE_MAIN_GUI_SCREEN_SIZE) {
+                    botSettings.mainGuiWidth = getWidth();
+                    botSettings.mainGuiHeight = getHeight();
+                    ConfigEntity.changed();
+                }
+            }
+        });
```

Comment:
Be aware that this will cause the config file to be updated every time you resize or move the window, i'd consider doing it on window closing